server/lib/chat/chat-api.ts

```diff
@@ -439,4 +463,182 @@ export class AdminChatApi {
 
     ctx.status = 204
   }
+
+  @httpGet('/banners/:bannerId')
+  @httpBefore(checkAllPermissions('manageChannelContent'))
+  async getChannelBanner(ctx: RouterContext): Promise<AdminGetChannelBannerResponse> {
+    const {
+      params: { bannerId },
+    } = validateRequest(ctx, {
+      params: Joi.object<{ bannerId: ChannelBannerId }>({
+        bannerId: Joi.string().uuid().required(),
+      }),
+    })
+
+    const channelBanner = await adminGetChannelBanner(bannerId)
+
+    if (!channelBanner) {
+      throw new ChatServiceError(
+        ChatServiceErrorCode.ChannelBannerNotFound,
+        'channel banner not found',
+      )
+    }
+
+    const channelIds = new Set<SbChannelId>()
+    for (let j = 0; j < channelBanner.availableIn.length; j++) {
+      channelIds.add(channelBanner.availableIn[j])
+    }
+    const channelInfos = await getChannelInfos(Array.from(channelIds))
+
+    return {
+      channelBanner: toChannelBannerJson(channelBanner),
+      channelInfos: channelInfos.map(c => toBasicChannelInfo(c)),
+    }
+  }
+
+  @httpGet('/banners')
+  @httpBefore(checkAllPermissions('manageChannelContent'))
+  async getChannelBanners(ctx: RouterContext): Promise<AdminGetChannelBannersResponse> {
+    const channelBanners = await adminGetChannelBanners()
+    const channelIds = new Set<SbChannelId>()
+    for (let i = 0; i < channelBanners.length; i++) {
+      for (let j = 0; j < channelBanners[i].availableIn.length; j++) {
+        channelIds.add(channelBanners[i].availableIn[j])
+      }
+    }
+    const channelInfos = await getChannelInfos(Array.from(channelIds))
+
+    return {
+      channelBanners: channelBanners.map(b => toChannelBannerJson(b)),
+      channelInfos: channelInfos.map(c => toBasicChannelInfo(c)),
+    }
+  }
+
+  @httpPost('/banners')
+  @httpBefore(checkAllPermissions('manageChannelContent'), handleMultipartFiles(MAX_IMAGE_SIZE))
+  async addChannelBanner(ctx: RouterContext): Promise<AdminUploadChannelBannerResponse> {
+    const {
+      body: { name, availableIn },
+    } = validateRequest(ctx, {
+      body: Joi.object<AdminUploadChannelBannerRequest & { image: any }>({
+        name: Joi.string().required(),
+        availableIn: Joi.array().items(Joi.string()),
+        image: Joi.any(),
+      }),
+    })
+
+    const imageFile = ctx.request.files?.image
+    if (!imageFile) {
+      throw new httpErrors.BadRequest('image file is required')
+    }
+    if (Array.isArray(imageFile)) {
+      throw new httpErrors.BadRequest('only one channel banner file can be uploaded')
+    }
+    const [image, imageExtension] = await resizeImage(
+      imageFile.filepath,
+      CHANNEL_BANNER_WIDTH,
+      CHANNEL_BANNER_HEIGHT,
+    )
+
+    return await transact(async client => {
+      const channelInfos =
+        availableIn && availableIn.length > 0 ? await findChannelsByName(availableIn) : []
```

Comment:
should probably dedupe these (unless `findChannelsByName` does that, but I don't think it does)